app/views/admin/api_docs/base/swagger.html.erb

```diff
@@ -37,18 +37,70 @@
       }
     }
 
+    function injectAuthParams(spec) {
+      var version = <%= @api_docs_service.service.proxy.authentication_method %>;
+      var authParams;
+      
+      switch(version) {
+        case 2:
+          authParams = [{
+            name: 'x-3scale-auth-app-key',
+            in: 'header',
+            description: 'App Key authentication parameter',
+            'x-data-three-scale-name': 'app_key',
+            required: true,
+            type: 'string'
+          },{
+            name: 'x-3scale-auth-app-id',
+            in: 'header',
+            description: 'App ID authentication parameter',
+            'x-data-three-scale-name': 'app_id',
+            required: true,
+            type: 'string'
+            }]
+          break;
+        
+        case 1:   
+          authParams = [{
+            name: 'x-3scale-auth-user-key',
+            in: 'header',
+            description: 'User Key authentication parameter',
+            'x-data-three-scale-name': 'user_key',
+            required: true,
+            type: 'string'
+          }];
+          break;
+            
+        default:
+        return;
+      }
+          
+      for (var p in spec.paths) {
+        var path = spec.paths[p];
+        for (var verb in path) {
+          Array.prototype.push.apply(path[verb].parameters, authParams);
```

Comment:
we should check that the parameter is not inside the spec already otherwise it will be added twice

Reply:
Very good point.

Reply:
and this also could be in a decorator, right?